### DISCUSSION TITLE
Need a way to monitor available mirth memory

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#6379] [Q&A]
Is there any way, using a mirth channel, to determine how much memory is available\used by Mirth connect and postgres? We recently moved from an appliance to an azure based install without an appliance, where we were able to monitor that information. We recently had an instance where we ran out of heap space without warning. Looking for a way to prevent this from happening.

**Comment by pacmano1** [comment]
I would not use a tool (mirth) to monitor itself with respect to its consumption of operating system resources.   What if the channel's call to mirth to check those things results in the out of memory condition?   I would think you want proper server monitoring tools installed.

**Comment by kirbykn2** [comment]
I agree with pacmano.  What kind of Azure-based install(server-windows or
Linux, container-ACI, Azure Kubernetes Service, other)? For your particular
install, what is the best solution for monitoring?  What does your team
have experience with?

On Tue, Dec 17, 2024 at 8:28 AM pacmano1 ***@***.***> wrote:

> I would not use a tool (mirth) to monitor itself with respect to its
> consumption of operating system resources. What if the channel's call to
> mirth to check those things results in the out of memory condition? I would
> think you want proper server monitoring tools installed.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nextgenhealthcare/connect/discussions/6379#discussioncomment-11593402>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APRXWD5UXWHVNUO6RF5YWQ32GARIXAVCNFSM6AAAAABTWT4VGGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNJZGM2DAMQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: <nextgenhealthcare/connect/repo-discussions/6379/comments/11593402@
> github.com>
>


-- 
Best,

Kirby Knight
| 231.735.4650 | ***@***.***

**Comment by jonbartels** [comment]
Mirth is a Java application. Use any monitoring tool that supports JMX.

**Reply by pacmano1** [reply]
Agreed for Mirth, but usually not adequate for full server/OS monitoring.

**Comment by rutledj** [comment]
Thanks for the info. Unfortunately, I think our devops\networking team is learning as they go with this mirth azure install. They are installed on windows servers. They will not give us access to the azure environment or mirth servers, so we (interface team) have no way to see what is happening or how close we are to running out of resources. I'll try to have a talk with that team to see if they know of any available tools for monitoring and alerting.
